File: go_src/21.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(l1 *ListNode, l2 *ListNode) *ListNode {
    r := &ListNode{}
    cur := r
    for l1 != nil && l2 != nil {
        if l1.Val < l2.Val {
            cur.Next = &ListNode{Val: l1.Val}
            l1 = l1.Next
        } else {
            cur.Next = &ListNode{Val: l2.Val}
            l2 = l2.Next
        }
        cur = cur.Next
    }
    for l1 != nil {
        cur.Next = &ListNode{Val: l1.Val}
        l1 = l1.Next
        cur = cur.Next
    }
    for l2 != nil {
        cur.Next = &ListNode{Val: l2.Val}
        l2 = l2.Next
        cur = cur.Next
    }
    return r.Next
}

File: java_src/21.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode head = new ListNode(-1);
        ListNode ptr = head;
        while (l1 != null && l2 != null) {
            if (l1.val < l2.val) {
                ptr.next = l1;
                l1 = l1.next;
            } else {
                ptr.next = l2;
                l2 = l2.next;
            }
            ptr = ptr.next;
        }

        while (l1 != null) {
            ptr.next = l1;
            l1 = l1.next;
            ptr = ptr.next;
        }
        while (l2 != null) {
            ptr.next = l2;
            l2 = l2.next;
            ptr = ptr.next;
        }

        return head.next;
    }
